app/domain/authentication/authn_jwt/jwt_configuration_factory.rb

```diff
@@ -0,0 +1,17 @@
+module Authentication
+  module AuthnJwt
+    # Factory that receives a vendor name and returns the appropriate JWT vendor configuration class
+    class JwtConfigurationFactory
+      VENDORS = {
+        "dummy" => JWTConfigurationDummyVendor
+      }
+
+      def create_jwt_configuration(vendor)
+        unless VENDORS.key?(vendor)
+          raise "Vendor #{vendor} not implemented yet."
+        end
+        (VENDORS[vendor]).new
```

Comment:
Is there a reason to call `new` every time?
If these classes does not maintain state (and I don't think they should), you can create them once (lazily, if you want), and pass on the same object every time.

Reply:
Changed the configuration function to static and removed new